examples/jax/run_maddpg_test.py

```diff
@@ -0,0 +1,72 @@
+# python3
+# Copyright 2021 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# TODO(Arnu): remove mypy ignore when the system is running
+# type: ignore
+
+"""Test for running Jax MADDPG on debug MPE environment."""
+from datetime import datetime
+
+import optax
+
+import mava.utils as utils
+from mava.systems.tf import maddpg
+
+
+def test_main() -> None:
+    """Acceptance test for Jax MADDPG system."""
+
+    # Create system and config
+    system = maddpg.MADDPG()
+
+    # Environment.
+    environment_factory = utils.make_factory(
```

Comment:
Is the assumption that we always use factories? I guess if we are single process, we probably don't need factories and should just return the envs, loggers and networks directly?

Reply:
Agreed. It sounds better that the user just provides the bare minimum, e.g. make_env fn, networks and args and then internally we add any needed complexities :slightly_smiling_face: But if factories are the bare minimum for it to look neat then we can keep it. We should at least not have different code here for single and multiple process.